packages/builder/src/stores/builder/components.js

```diff
@@ -477,7 +552,8 @@ export class ComponentStore extends BudiStore {
       // Mutates the fetched component with updates
       const patchResult = patchFn(component, screen)
 
-      // Mutates the component with any required settings updates
+      // Post processing
+      this.enrichEmptySettings(component, { screen, useDefaultValues: false })
```

Comment:
I've added `enrichEmptySettings` inside the core component patch function. This means that after any change to a component, we'll attempt to enrich empty settings again (rather than only on creation, which we do now).
This has a few benefits - the main one being we can now add more logic (like this cards block stuff) in here and ensure that it will be ran again if required. This works well in combination with the "resetOn" manifest key. Use that key to wipe settings, then this enrichment will pick them up and default them again.

Reply:
This behaviour seems to have unpredictable results when clearing the card block values. If I delete `Title`, `Subtitle` and `Description`, sometimes all 3 values will remain empty but in some instances maybe 2 values will regenerate. When I deleted all 3 values, some internal defaults appeared and some old config remained. As below:

![Screenshot 2024-05-20 at 16 19 41](https://github.com/Budibase/budibase/assets/5913006/d96679a5-3fc8-4ac6-ae80-57ab03561d22)

Reply:
This seems to just be a race condition thing depending on what field you clear last. The last field you clear will trigger the enrichment and all the settings do get filled back in, but it looks like the blur of the final field sometimes clears whatever you have selected again. 
I'll see if I can put in something to prevent that happening.

Reply:
Following up on this again - it's an existing issue with any component where if a setting value is `foo`, and you clear it, then empty setting enrichment sets the value back to `foo`, svelte sees that the value is identical and so does not re-render the setting with the same value, yet the input itself has actually been cleared. So you end up in this weird position where the saved value does exist but the input is empty.
Not entirely sure how to solve this without doing something nuclear like key the entire component setting section by screen rev. It's very much an edge case though as it depends on clearing the exact value that will be repopulated by enrichment.

Reply:
I see what you mean now. Something to keep in mind but in no way a new thing 👍